File: GBEmu/GBEmu.Emulation/Flags.cs
using System;

namespace GBEmu.Emulation
{
	[Flags]
	public enum Flags : byte
	{
		/// <summary>
		/// Zero
		/// </summary>
		Z = 1 << 7, //Zero
		/// <summary>
		/// Subtract
		/// </summary>
		N = 1 << 6, //Subtract
		/// <summary>
		/// HalfCarry
		/// </summary>
		H = 1 << 5, //HalfCarry
		/// <summary>
		/// Carry
		/// </summary>
		C = 1 << 4  //Carry
	}
}

File: GBEmu/GBEmu/Flags.swift
//
//  Flags.swift
//  GBEmu
//
//  Created by Maurus Kühne on 13.11.14.
//  Copyright (c) 2014 Maurus Kühne. All rights reserved.
//

import Foundation

enum Flags : UInt8 {
  case Zero =       0b10000000
  case Subtract =   0b01000000
  case HalfCarry =  0b00100000
  case Carry =      0b00010000
  
  var description: String {
    get {
      switch(self) {
      case .Zero :
        return "Zero"
      case .Subtract :
        return "Subtract"
      case .Carry :
        return "Carry"
      case .HalfCarry :
        return "HalfCarry"
      }
    }
  }
}